exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -413,4 +413,8 @@
 
   String DYNAMIC_UDF_SUPPORT_ENABLED = "exec.udf.enable_dynamic_support";
   BooleanValidator DYNAMIC_UDF_SUPPORT_ENABLED_VALIDATOR = new BooleanValidator(DYNAMIC_UDF_SUPPORT_ENABLED, true, true);
+
+  String ZK_QUERY_STATE_UPDATE_KEY = "drill.exec.zk.query.state.update";
```

Comment:
Maybe ENABLE_QUERY_STATE_UPDATE_KEY? For the key name, maybe don't include ZK. ZK is our solution today, but this option would be equally valid if we used some other mechanism.